packages/block-renderer/src/hooks/use-pattern-inline-css.tsx

```diff
@@ -0,0 +1,41 @@
+// Memoize order per pattern to show same order in previews
+const inlineCssByPatternId: { [ key: string ]: string } = {};
+
+// Memoize last offset
+let lastOffset = 0;
+
+const usePatternInlineCss = (
+	patternId: string,
+	patternHtml: string,
+	shouldShufflePosts: boolean
+) => {
+	const hasGrid = patternHtml?.includes( 'is-layout-grid' );
+	const blogPostCount = patternHtml?.match( /wp-block-post /g )?.length ?? 0;
+
+	// Only for patterns with a grid of posts in newly created sites
+	if ( ! shouldShufflePosts || ! hasGrid || blogPostCount <= 1 ) {
+		return undefined;
+	}
+
+	// Return memoized order
+	let inlineCss = inlineCssByPatternId[ patternId ] || '';
+	if ( inlineCss ) {
+		return inlineCss;
+	}
+
+	// Create CSS rules
+	[ ...Array( blogPostCount ).keys() ].forEach( ( order, index ) => {
+		const childIndex = index + 1;
+		// Offset order of blog posts to avoid repetition in previews
+		const orderWithOffset = ( ( order - lastOffset + blogPostCount ) % blogPostCount ) + 1;
```

Comment:
Sorry @miksansegundo I made a slight mistake; the formula should be like this (otherwise the order could be negative)

```
const offset = lastOffset % blogPostCount;
const orderWithOffset = ( ( order - offset + blogPostCount ) % blogPostCount ) + 1;
```

I could not simplify it any further 😅

Reply:
What could be the issue if we have a negative [order](https://developer.mozilla.org/en-US/docs/Web/CSS/order)?

I mean, maybe it's a good thing because will make it more random.

Reply:
TIL that the orders are relative; I thought the orders must be numbers between 1 and `blogPostCount`.

> What could be the issue

There is no issue, but then the offset is "slightly mathematically wrong" :) as it is intended to be a "cyclic" offset (between 1 and `blogPostCount`). E.g. for the following example:

<img width="544" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/fd85805e-7939-4880-9d77-2c50c98e347a">

The first two posts should appear last, not first, according to the original intention of the offset.

But maybe it does not matter much, so I'll leave it to you!

P.S.: also just realized in any case we can omit ` + 1` at the end of `orderWithOffset` because the order is relative anyway.

Reply:
Thanks for the details. I updated the formula and the results still look good 👍